src/main/scala/com/wavesplatform/history/CheckpointStorage.scala

```diff
@@ -0,0 +1,35 @@
+package com.wavesplatform.history
+
+import org.h2.mvstore.{MVMap, MVStore}
+import scorex.network.{BlockCheckpoint, Checkpoint}
+import scorex.transaction.CheckpointService
+import scorex.utils.LogMVMapBuilder
+
+trait CheckpointStorage {
+  def checkpoint: java.util.Map[Int, (Seq[(Int, Array[Byte])], Array[Byte])]
+
+  def commit(): Unit
+}
+
+class MVStoreCheckpointStorage(db: MVStore) extends CheckpointStorage {
+  val checkpoint: MVMap[Int, (Seq[(Int, Array[Byte])], Array[Byte])] = db.openMap("checkpoint", new LogMVMapBuilder[Int, (Seq[(Int, Array[Byte])], Array[Byte])])
+
+  def commit(): Unit = db.commit()
+}
+
+class CheckpointServiceImpl(storage: CheckpointStorage) extends CheckpointService {
```

Comment:
There doesn't seem to be any other implementations of `CheckpointStorage` except for `MVStoreCheckpointStorage`, and there are no tests where `CheckpointServiceImpl`'s functionality is tested with a fake instance of `CheckpointStorage`. My point is, instead of three entities, why not have just one
```scala
class CheckpointServiceImpl(db: MVStore) {
```